x/concentrated-liquidity/position.go

```diff
@@ -463,49 +469,41 @@ func (k Keeper) fungifyChargedPosition(ctx sdk.Context, owner sdk.AccAddress, po
 		return 0, err
 	}
 
+	// Compute the fee growth outside of the range between lower tick and upper tick
+	feeGrowthOutside, err := k.getFeeGrowthOutside(ctx, poolId, lowerTick, upperTick)
+	if err != nil {
+		return 0, err
+	}
+
 	// Move unclaimed rewards from the old positions to the new position.
 	// Also, delete the old positions from state.
 
 	// Loop through each position ID.
-	for _, positionId := range positionIds {
+	for _, oldPositionId := range positionIds {
 		// Loop through each uptime accumulator for the pool.
 		for uptimeIndex, uptimeAccum := range uptimeAccumulators {
-			oldPositionName := string(types.KeyPositionId(positionId))
+			oldPositionName := string(types.KeyPositionId(oldPositionId))
 			// Check if the accumulator contains the position.
 			hasPosition, err := uptimeAccum.HasPosition(oldPositionName)
 			if err != nil {
 				return 0, err
 			}
 			// If the accumulator contains the position, move the unclaimed rewards to the new position.
 			if hasPosition {
-				// Get number of share in the position.
-				positionShares, err := uptimeAccum.GetPositionSize(oldPositionName)
-				if err != nil {
-					return 0, err
-				}
-
-				// Remove the position from the accumulator so that it is deleted when
-				// rewards are claimed.
-				if err := uptimeAccum.RemoveFromPosition(oldPositionName, positionShares); err != nil {
-					return 0, err
-				}
-
-				// Prepare the accumulator for the old position
-				rewards, dust, err := prepareAccumAndClaimRewards(uptimeAccum, oldPositionName, uptimeGrowthOutside[uptimeIndex])
-				if err != nil {
-					return 0, err
-				}
-				unclaimedRewardsForPosition := sdk.NewDecCoinsFromCoins(rewards...).Add(dust...)
-
-				// Add the unclaimed rewards to the new position.
-				err = uptimeAccum.AddToUnclaimedRewards(newPositionName, unclaimedRewardsForPosition)
-				if err != nil {
+				if err := moveRewardsToNewPositionAndDeleteOld(ctx, uptimeAccum, oldPositionName, newPositionUptimeAccName, uptimeGrowthOutside[uptimeIndex]); err != nil {
 					return 0, err
 				}
 			}
 		}
+
+		// Move fees into the new fee accumulator.
+		oldPositionFeeName := types.KeyFeePositionAccumulator(oldPositionId)
+		if err := moveRewardsToNewPositionAndDeleteOld(ctx, feeAccumulator, oldPositionFeeName, newPositionFeeAccName, feeGrowthOutside); err != nil {
+			return 0, err
```

Comment:
Hmm I don't think there is a need for testing internal methods at higher levels of abstraction. The unit test for `moveRewardsToNewPositionAndDeleteOld` already does this.

In addition, the desired behavior around this is checked and guaranteed by:
https://github.com/osmosis-labs/osmosis/blob/4c0b2a90d7c6ea24f3c6230c01235348e57750f6/x/concentrated-liquidity/position_test.go#L1126-L1131

The reason we cannot claim is because there are no accumulators